### ISSUE TITLE
bootstrap-local.js is replacing installed NodeJS built-in modules with old versions

### ISSUE THREAD

**Issue by paisanousa** [angular/angular-cli#9229]
### Versions

```
:~$ node -v
v9.3.0
:~$ npm -v
5.6.0
:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.3 LTS
Release:	16.04
Codename:	xenial

@angular/cli version (following clone/install of @angular/cli detailed in Repro steps):
:~/demo/myapp (master)$ ng -v

    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
    
Angular CLI: local (v1.7.0-beta.0, branch: master)
Node: 9.3.0
OS: linux x64
Angular: 5.2.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.7.0-beta.0
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.48
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.10.0-beta.0
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.5.3
webpack: 3.10.0
```
### Repro steps

```
Beginning list of global installs:
$ npm list --depth=0 -g
/home/dennis/.nvm/versions/node/v9.3.0/lib
└── npm@5.6.0
```
```
* Step 1 - Clone and globally symlink the angular/angular-cli github repository:

Using the documentation provided at:
https://github.com/angular/angular-cli#development-hints-for-working-on-angular-cli

Clone and link the angular/angular-cli repository:

:~$ git clone https://github.com/angular/angular-cli.git
Cloning into 'angular-cli'...
remote: Counting objects: 26358, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 26358 (delta 2), reused 2 (delta 0), pack-reused 26350
Receiving objects: 100% (26358/26358), 6.87 MiB | 465.00 KiB/s, done.
Resolving deltas: 100% (14924/14924), done.

:~$ cd angular-cli
:~/angular-cli (master)$ npm link
...
(skip install details)
...
added 1094 packages in 208.01s
/home/dennis/.nvm/versions/node/v9.3.0/bin/ng -> /home/dennis/.nvm/versions/node/v9.3.0/lib/node_modules/@angular/cli/bin/ng
/home/dennis/.nvm/versions/node/v9.3.0/lib/node_modules/@angular/cli -> /home/dennis/angular-cli

Verify global syslink:
:~/angular-cli (master)$ npm list --depth=0 -g
/home/dennis/.nvm/versions/node/v9.3.0/lib
├── @angular/cli@1.7.0-beta.0 -> /home/dennis/angular-cli
└── npm@5.6.0

npm ERR! peer dep missing: @angular-devkit/schematics@0.0.43, required by @schematics/angular@0.1.12
npm ERR! missing: cuint@latest, required by xxhashjs@0.2.1
```
```
* Step 2 - Create and serve new angular app using angular-cli:

Build angular app in demo directory:
:~/angular-cli (master)$ cd ..
:~$ mkdir demo
:~$ cd demo
:~/demo$ ng new myapp
...
(skip install details)
...
added 1300 packages in 96.341s
Installed packages for tooling via npm.
Successfully initialized git.
Project 'myapp' successfully created.

Create/verify local symlink:
:~/demo$ cd myapp
:~/demo/myapp (master)$ npm link @angular/cli
/home/dennis/demo/myapp/node_modules/@angular/cli -> /home/dennis/.nvm/versions/node/v9.3.0/lib/node_modules/@angular/cli -> /home/dennis/angular-cli

Verify current angular version in use:
:~/demo/myapp (master)$ ng -v

    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
    
Angular CLI: local (v1.7.0-beta.0, branch: master)
Node: 9.3.0
OS: linux x64
Angular: 5.2.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.7.0-beta.0
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.48
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.10.0-beta.0
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.5.3
webpack: 3.10.0

:~/demo/myapp (master)$ ng serve
** NG Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
...
(skip spin-up details)
...
absolute-iDate: 2018-01-15T20:55:12.484Z                
Hash: cef1f45efe3d382cfc48
Time: 11516ms
chunk {inline} inline.bundle.js (inline) 5.79 kB [entry] [rendered]
chunk {main} main.bundle.js (main) 723 kB [initial] [rendered]
chunk {polyfills} polyfills.bundle.js (polyfills) 569 kB [initial] [rendered]
chunk {styles} styles.bundle.js (styles) 51.7 kB [initial] [rendered]
chunk {vendor} vendor.bundle.js (vendor) 6.71 MB [initial] [rendered]

webpack: Compiled successfully.
```
```
* Step 3 - Open browser and navigate to http://localhost:4200/

Select browser, in this instance: Google Chrome Version 63.0.3239.132 (Official Build) (64-bit) 
Navigate to http://localhost:4200/

(subsequent behavior recorded below)
```

### Observed behavior

```
Note: In addition to the Chrome browser noted above, the following behavior has been observed 
      in Firefox Quantum Version 57.0.4 (64-bit) browser. This behavior has also been observed 
      using a MacBook Pro running macOS High Sierra Version 10.13.2 in three browsers: Chrome, 
      Firefox and Safari.
```
```
Immediately following the initial server response, the error below is displayed in the browser 
and at the server. The server remains online, and the error will re-occur if the browser is 
reloaded. The error will continue to re-occur if other windows or browsers attempt the 
navigation.
  
TypeError: querystring.unescape is not a function
    at getFilenameFromUrl (/home/dennis/demo/myapp/node_modules/webpack-dev-middleware/lib/GetFilenameFromUrl.js:33:21)
    at module.exports (/home/dennis/demo/myapp/node_modules/webpack-dev-middleware/lib/GetFilenameFromUrl.js:62:9)
    at webpackDevMiddleware (/home/dennis/demo/myapp/node_modules/webpack-dev-middleware/middleware.js:40:18)
    at Layer.handle [as handle_request] (/home/dennis/demo/myapp/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:317:13)
    at /home/dennis/demo/myapp/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:335:12)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:275:10)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:127:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at Server.setContentHeaders (/home/dennis/demo/myapp/node_modules/webpack-dev-server/lib/Server.js:502:3)
    at Layer.handle [as handle_request] (/home/dennis/demo/myapp/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:137:13)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at Route.dispatch (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/dennis/demo/myapp/node_modules/express/lib/router/layer.js:95:5)
    at /home/dennis/demo/myapp/node_modules/express/lib/router/index.js:281:22
    at param (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:354:14)
    at param (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:410:3)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:275:10)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:127:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at Server.app.all (/home/dennis/demo/myapp/node_modules/webpack-dev-server/lib/Server.js:79:47)
    at Layer.handle [as handle_request] (/home/dennis/demo/myapp/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:137:13)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:131:14)
    at Route.dispatch (/home/dennis/demo/myapp/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/dennis/demo/myapp/node_modules/express/lib/router/layer.js:95:5)
    at /home/dennis/demo/myapp/node_modules/express/lib/router/index.js:281:22
    at param (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:354:14)
    at param (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:410:3)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:275:10)
    at expressInit (/home/dennis/demo/myapp/node_modules/express/lib/middleware/init.js:40:5)
    at Layer.handle [as handle_request] (/home/dennis/demo/myapp/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:317:13)
    at /home/dennis/demo/myapp/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:335:12)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:275:10)
    at query (/home/dennis/demo/myapp/node_modules/express/lib/middleware/query.js:45:5)
    at Layer.handle [as handle_request] (/home/dennis/demo/myapp/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:317:13)
    at /home/dennis/demo/myapp/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:335:12)
    at next (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:275:10)
    at Function.handle (/home/dennis/demo/myapp/node_modules/express/lib/router/index.js:174:3)
    at Function.handle (/home/dennis/demo/myapp/node_modules/express/lib/application.js:174:10)
    at Server.app (/home/dennis/demo/myapp/node_modules/express/lib/express.js:39:9)
    at Server.new_handler (/home/dennis/demo/myapp/node_modules/sockjs/lib/utils.js:89:20)
    at Server.emit (events.js:159:13)
    at parserOnIncoming (_http_server.js:619:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:115:23)
```
### Desired behavior

```
The expected behavior would be a browser display of the Welcome title, Angular logo, and info 
links that are normally seen when using the globally installed @angular/cli package and building 
an application using the command "ng new myapp".
```
### Mention any other details that might be useful (optional)

The observed error occurs because bootstrap-local.js replaces the installed version of the NodeJS
built-in module 'querystring' with an older (and possibly customized) version found in the local 
node_modules folder.  Other built-in modules are similarly being replaced, but have no apparent
influence on this particular error.  In my testing, I have found there is another error waiting to 
occur behind this one. That error is caused by a similar substitution of the 'buffer' module.

The replacement of 'querystring' is a direct result of the modifications applied to bootstrap-local.js 
in commit dd9e69e . This finding can be verified by commenting out the code added by the commit,
press save, stop/start the app server, and navigate to localhost:4200. The browser display will be 
produced as expected and without error.

If needed, I am prepared to submit a pull request that resolves this issue by excluding NodeJS 
built-ins from the group of modules affected by the changes in commit dd9e69e .

Also, I can provide additional debugging information to confirm and/or resolve this issue.

**Comment by paisanousa** [comment]
Pull request #9302 submitted.

**Comment by paisanousa** [comment]
Merged PR #9302 resolved this issue.
